src/util/customTransformer-faas.js

```diff
@@ -0,0 +1,104 @@
+const { v4: uuidv4 } = require("uuid");
+const stats = require("./stats");
+const { getMetadata } = require("../v0/util");
+const { setupFaasFunction, executeFaasFunction } = require("./openfaas");
+
+function generateFunctionName(userTransformation, testMode) {
+  if (testMode) {
+    const funcName = `fn-test-${userTransformation.testName?.replace(
+      "_",
+      "-"
+    )}-${uuidv4()}`;
+    return funcName.substring(0, 63).toLowerCase();
+  }
+
+  return `fn-${userTransformation.workspaceId}-${userTransformation.versionId}`
+    .substring(0, 63)
+    .toLowerCase();
+}
+
+async function setOpenFaasUserTransform(
+  userTransformation,
+  testWithPublish,
+  pregeneratedFnName,
+  testMode = false
+) {
+  if (!testWithPublish) {
+    return { success: true };
+  }
+
+  const tags = {
+    transformerVersionId: userTransformation.versionId,
+    language: userTransformation.language,
+    identifier: "openfaas",
+    publish: testWithPublish
+  };
+  const functionName =
+    pregeneratedFnName || generateFunctionName(userTransformation, testMode);
+  const setupTime = new Date();
+
+  await setupFaasFunction(
+    functionName,
+    userTransformation.code,
+    userTransformation.versionId,
+    testMode
+  );
+
+  stats.timing("faas_publish_time", setupTime, tags);
```

Comment:
We already have a:
```
    identifier: "openfaas",
```
In the tags. Would it make sense to rename the metric to `transformation_creation_time`?

Reply:
By default `transformer_` gets prefixed to stat.
Named it `creation_time`